build-tools/create-packs/Directory.Build.props

```diff
@@ -13,6 +13,8 @@
     <SuppressDependenciesWhenPacking>true</SuppressDependenciesWhenPacking>
     <!-- Ignore TargetFramework reference group related warnings, these are workload packs not functional NuGets. -->
     <NoWarn>$(NoWarn);NU5128;NU5131</NoWarn>
+    <!-- Allow NU5123 for pull requests, which will have $SYSTEM_PULLREQUEST_ISFORK set to true or false -->
+    <NoWarn Condition=" '$(SYSTEM_PULLREQUEST_ISFORK)' != '' ">$(NoWarn);NU5123</NoWarn>
```

Comment:
I'm not sure why, but this was the only `$SYSTEM_PULLREQUEST_*` variable set:

* https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables

The rest were blank.

Reply:
Dovetailing with [this question](https://github.com/dotnet/android/pull/9983#issuecomment-2770522781), what is the relevance to `$(SYSTEM_PULLREQUEST_ISFORK)` to ignore NU5123?  Why is it (seemingly) safe to ignore NU5123 on PR builds, but not on "local"/"on my machine" builds?

Reply:
I was thinking it would be good to allow `NU5123` to warn (and then error) on release branches.

I don't believe any `$(SYSTEM_PULLREQUEST_*)` variables are set on release branches, because it isn't a pull request.

Reply:
Unfortunately `$(SYSTEM_PULLREQUEST_ISFORK)` appears to always be set, it's present and set to False when running a CI build against a main branch.

Reply:
@pjcollins is there any other way to know it's a pull request? Or do we just need to `$(NoWarn)=NU5123` no matter what?

Reply:
Ok, I see both these env vars on one of the older builds:

![image](https://github.com/user-attachments/assets/3583818b-127e-45b8-9af1-c96c8d4f03df)